addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js

```diff
@@ -328,7 +327,13 @@ class EnvironmentScService extends Service {
     return _.filter(envs, env => !_.includes(filterStatus, env.status) && !env.status.includes('FAILED'));
   }
 
-  async find(requestContext, { id, fields = [], fetchCidr = true }) {
+  async find(requestContext, { id, fields = [], fetchCidr }) {
+    // Define fetchCidr flag based on isAppStreamEnabled config rather than defaulting it to true
+    const isAppStreamEnabled = this.isAppStreamEnabled();
+    const computedFetchCidr = fetchCidr === undefined ? !isAppStreamEnabled : fetchCidr;
+    if (computedFetchCidr && isAppStreamEnabled) {
```

Comment:
Can we simplify lines 333 and 334 to something like:
```
if (fetchCidr && isAppStreamEnabled)
```

Reply:
I do need a new fetchCidr field which is derived from already provided fetchCidr and isAppStreamEnabled flag. Note that value gets used before calling `getSecurityGroupDetails`.